tiledb/sm/filesystem/s3.h

```diff
@@ -80,14 +81,287 @@
+      , aws_external_id_(config.get<std::string>(
+            "vfs.s3.aws_external_id", Config::must_find))
+      , aws_load_frequency_(config.get<std::string>(
+            "vfs.s3.aws_load_frequency", Config::must_find))
+      , aws_session_name_(
+            config.get<std::string>("vfs.s3.aws_session_name").value())
+      , scheme_(config.get<std::string>("vfs.s3.scheme", Config::must_find))
+      , endpoint_override_(config.get<std::string>(
+            "vfs.s3.endpoint_override", Config::must_find))
+      , use_virtual_addressing_(config.get<bool>(
+            "vfs.s3.use_virtual_addressing", Config::must_find))
+      , skip_init_(config.get<bool>("vfs.s3.skip_init", Config::must_find))
+      , use_multipart_upload_(
+            config.get<bool>("vfs.s3.use_multipart_upload", Config::must_find))
+      , max_parallel_ops_(
+            config.get<uint64_t>("vfs.s3.max_parallel_ops", Config::must_find))
+      , multipart_part_size_(config.get<uint64_t>(
+            "vfs.s3.multipart_part_size", Config::must_find))
+      , connect_timeout_ms_(
+            config.get<int64_t>("vfs.s3.connect_timeout_ms", Config::must_find))
+      , connect_max_tries_(
+            config.get<int64_t>("vfs.s3.connect_max_tries", Config::must_find))
+      , connect_scale_factor_(config.get<int64_t>(
+            "vfs.s3.connect_scale_factor", Config::must_find))
+      , logging_level_(
+            config.get<std::string>("vfs.s3.logging_level", Config::must_find))
+      , request_timeout_ms_(
+            config.get<int64_t>("vfs.s3.request_timeout_ms", Config::must_find))
+      , requester_pays_(
+            config.get<bool>("vfs.s3.requester_pays", Config::must_find))
+      , proxy_host_(
+            config.get<std::string>("vfs.s3.proxy_host", Config::must_find))
+      , proxy_port_(
+            config.get<uint32_t>("vfs.s3.proxy_port", Config::must_find))
+      , proxy_scheme_(
+            config.get<std::string>("vfs.s3.proxy_scheme", Config::must_find))
+      , proxy_username_(
+            config.get<std::string>("vfs.s3.proxy_username", Config::must_find))
+      , proxy_password_(
+            config.get<std::string>("vfs.s3.proxy_password", Config::must_find))
+      , no_sign_request_(
+            config.get<bool>("vfs.s3.no_sign_request", Config::must_find))
+      , sse_algorithm_(config.get<std::string>("vfs.s3.sse", Config::must_find))
+      , sse_kms_key_id_(
+            sse_algorithm_ == "kms" ?
+                config.get<std::string>("vfs.s3.sse_kms_key_id").value() :
+                "")
+      , bucket_acl_str_(config.get<std::string>(
+            "vfs.s3.bucket_canned_acl", Config::must_find))
+      , object_acl_str_(config.get<std::string>(
+            "vfs.s3.object_canned_acl", Config::must_find))
+      , config_source_(config.get<std::string>(
+            "vfs.s3.config_source", Config::must_find)){};
+
+  ~S3Parameters() = default;
+
+  /**
+   * The AWS region.
+   * Default: us-east-1
```

Comment:
I would remove the defaults here and make all the comments single line when possible. The defaults are already duplicated in the set configuration documentation and I think we need to prevent a third place to update that will be different depending on where the settings live.

Reply:
cc @eric-hughes-tiledb 
Eric and I discussed this offline. The idea was that the config declarations will eventually be fully moved out of `Config` and into the `ConfigParams` structs (per notes L91-94 in this file). 
I'm fine with reverting this change as long as we all agree on it

Reply:
Note: reverting these changes to be addressed in future work